test/e2e/major_upgrade_v2.go

```diff
@@ -0,0 +1,142 @@
+package main
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log"
+	"strings"
+	"time"
+
+	"github.com/celestiaorg/celestia-app/v2/app"
+	"github.com/celestiaorg/celestia-app/v2/app/encoding"
+	v1 "github.com/celestiaorg/celestia-app/v2/pkg/appconsts/v1"
+	v2 "github.com/celestiaorg/celestia-app/v2/pkg/appconsts/v2"
+	"github.com/celestiaorg/celestia-app/v2/test/e2e/testnet"
+	"github.com/celestiaorg/celestia-app/v2/test/txsim"
+	"github.com/celestiaorg/knuu/pkg/knuu"
+	"github.com/tendermint/tendermint/rpc/client/http"
+)
+
+func MajorUpgradeToV2(logger *log.Logger) error {
+	latestVersion, err := testnet.GetLatestVersion()
+	testnet.NoError("failed to get latest version", err)
+
+	logger.Println("Running major upgrade to v2 test", "version", latestVersion)
+
+	numNodes := 4
+	upgradeHeight := int64(12)
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	logger.Println("Creating testnet")
+	testNet, err := testnet.New("runMajorUpgradeToV2", seed, nil, "test")
+	testnet.NoError("failed to create testnet", err)
+
+	defer testNet.Cleanup()
+
+	testNet.SetConsensusParams(app.DefaultInitialConsensusParams())
+
+	preloader, err := knuu.NewPreloader()
+	testnet.NoError("failed to create preloader", err)
+
+	defer func() { _ = preloader.EmptyImages() }()
+	testnet.NoError("failed to add image", preloader.AddImage(testnet.DockerImageName(latestVersion)))
+
+	logger.Println("Creating genesis nodes")
+	for i := 0; i < numNodes; i++ {
+		err := testNet.CreateGenesisNode(latestVersion, 10000000, upgradeHeight, testnet.DefaultResources)
+		testnet.NoError("failed to create genesis node", err)
+	}
+
+	kr, err := testNet.CreateAccount("alice", 1e12, "")
+	testnet.NoError("failed to create account", err)
+	// start the testnet
+
+	logger.Println("Setting up testnet")
+	testnet.NoError("Failed to setup testnet", testNet.Setup())
+	logger.Println("Starting testnet")
+	testnet.NoError("Failed to start testnet", testNet.Start())
+
+	errCh := make(chan error)
+	encCfg := encoding.MakeConfig(app.ModuleEncodingRegisters...)
+	opts := txsim.DefaultOptions().WithSeed(seed).SuppressLogs()
+	sequences := txsim.NewBlobSequence(txsim.NewRange(200, 4000), txsim.NewRange(1, 3)).Clone(5)
+	sequences = append(sequences, txsim.NewSendSequence(4, 1000, 100).Clone(5)...)
+	go func() {
+		errCh <- txsim.Run(ctx, testNet.GRPCEndpoints()[0], kr, encCfg, opts, sequences...)
+	}()
+
+	// assert that the network is initially running on v1
```

Comment:
this comment seems placed incorrectly because the contents of the for loop below actually assert that the network is on v1 for certain headers and v2 for other headers. Proposal to get rid of the comment entirely

```suggestion
```